Flash Chat iOS13/Controllers/ChatViewController.swift

```diff
@@ -89,21 +89,42 @@ class ChatViewController: UIViewController, UITableViewDelegate, UITableViewData
 		if let messageBody = messageTextfield?.text,
 		   let messageSender = Auth.auth().currentUser?.email {
 			let currentDate = Date()
-			let data: [String : Any] = [
+			guard let selectedThread = selectedThread else { return }
+			let dataForMessage: [String : Any] = [
 				Constants.FStore.senderField : messageSender,
 				Constants.FStore.bodyField : messageBody,
 				Constants.FStore.dateField : currentDate
 			]
-			database.collection(Constants.FStore.threadsCollectionName).document((selectedThread?.idString)!).collection(Constants.FStore.bubblesField).addDocument(data: data) { [self] error in
+			database.collection(Constants.FStore.threadsCollectionName).document((selectedThread.idString)).collection(Constants.FStore.bubblesField).addDocument(data: dataForMessage) { [self] error in
 				if let error = error {
 					AppDelegate.showError(error, inViewController: self)
 				} else {
-					database.collection(Constants.FStore.threadsCollectionName).document((selectedThread?.idString)!).updateData([Constants.FStore.dateField : Date()])
+					database.collection(Constants.FStore.threadsCollectionName).document((selectedThread.idString)).updateData([Constants.FStore.dateField : Date()])
 					DispatchQueue.main.async { [self] in
 						messageTextfield?.text?.removeAll()
 					}
 				}
 			}
+			if selectedThread.recipients == [messageSender, messageSender] {
+				DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) { [self] in
+				let currentDatePlusAFewSeconds = Date()
+				let dataForSelfMessageEcho: [String : Any] = [
```

Comment:
💬 
Optional: might be helpful to add a comment to designate this block of code as specific handling of self-messaging, since it's not immediately clear